### PR TITLE
[M1] Webpack/Typescript/Angular

### PR DESCRIPTION
**Pull request by annapopova1**
Added project, initial components, interfaces.

### CODE REVIEW

File: src/app/models/course.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  description!: string;
+
+  constructor(title: string, description: string, duration: number) {
+    this.id = Math.random().toString(20).substr(2, 5);
+    this.title = title;
+    this.description = description;
+    this.duration = duration;
+    this.createDate = new Date();
+  }
+}
```
Comment (etray7) [inline src/app/models/course.ts]: As I see there is no sense to describe `class` like this, you can just implement it like: 
```
export class CourseMock implements Course {
  readonly id!: string;
  createDate!: Date;

  constructor(
    public title: string,
    public description: string, 
    public duration: number
  ) {
    this.id = Math.random().toString(20).substr(2, 5);
    this.createDate = new Date();
  }
}
```
When you add a modifier inside constructor arguments, it automatically assigns them under the hood, for example: `constructor(public title: string) {}` it is the same as `constructor(title: string) {this.title = title}`

File: src/app/models/user.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+export class UserMock implements User {
+  readonly id!: string;
+  readonly firstName!: string;
+  readonly lastName!: string;
+  
+  constructor(firstName: string, lastName: string) {
+    this.id = Math.random().toString(20).substr(2, 5);
+    this.firstName = firstName;
+    this.lastName = lastName;
+  }
```
Comment (etray7) [inline src/app/models/user.ts]: Same as for Course

**Review by etray7 — COMMENTED** [review]
As well, I recommend you to read about folder/module style guide: https://angular.io/guide/styleguide#overall-structural-guidelines, but for this simple project you can leave it as it is :)

File: src/app/models/course.ts (hunk 1 above)
Comment (etray7) [reply src/app/models/course.ts]: If you want to use the `!` operator for non-null checks, you can leave your own option, but I think that in this case there is no need

Comment (annapopova1) [reply src/app/models/course.ts]: Good to know. Thanks.

**Comment by annapopova1** [comment]
> As well, I recommend you to read about folder/module style guide: https://angular.io/guide/styleguide#overall-structural-guidelines, but for this simple project you can leave it as it is :)

I tried to restructure the app a little bit. @etray7 could you please take a look one more time?

**Pull request merged**